backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/audit/ApiAuditLoggable.java

```diff
@@ -77,17 +79,17 @@ public ILoggable addLog(final String msg, final Object... msgParameters)
 	{
 		final ApiRequestAuditLog logEntry = createLogEntry(msg, msgParameters);
 
-		List<ApiRequestAuditLog> buffer = this.buffer;
-		if (buffer == null)
-		{
-			buffer = this.buffer = new ArrayList<>(bufferSize);
-		}
-		buffer.add(logEntry);
+		addToBuffer(logEntry);
 
-		if (buffer.size() >= bufferSize)
-		{
-			flush();
-		}
+		return this;
+	}
+
+	@Override
+	public ILoggable addTableRecordReferenceLog(final @NonNull ITableRecordReference recordRef, final @NonNull String type, final @NonNull String trxName)
```

Comment:
i think it makes sense to call this method with `StateType` isntead of `String`

Reply:
I agree, but I can't use StateType in ILoggable as it implies a circular dependency with `de.metas.adempiere.adempiere.base`.